src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -568,14 +572,18 @@ void MeshForwarder::HandleDataRequest(const Mac::Address &aMacSource, const otTh
     child->ResetLinkFailures();
     indirectMsgCount = child->GetIndirectMessageCount();
 
+#if OPENTHREAD_CONFIG_ENABLE_VERIFY_ACK_FP_FLAG
+    VerifyOrExit(aFrame.IsAckWithFramePending());
+#endif
+
     if (!mSourceMatchController.IsEnabled() || (indirectMsgCount > 0))
     {
         child->SetDataRequestPending(true);
     }
 
     mScheduleTransmissionTask.Post();
 
-    otLogInfoMac("Rx data poll, src:0x%04x, qed_msgs:%d, rss:%d", child->GetRloc16(), indirectMsgCount, aLinkInfo.mRss);
+    otLogDebgMac("Rx data poll, src:0x%04x, qed_msgs:%d, rss:%d", child->GetRloc16(), indirectMsgCount, aLinkInfo.mRss);
```

Comment:
Let's keep this log line as info level. This has been a very useful log line in investigating/debugging issues.
Also suggest, we update the log to indicate if acked with frame pending or not.

Reply:
Good point, added!